encode_number.go

```diff
@@ -143,10 +143,21 @@ func (e *Encoder) EncodeInt(n int64) error {
 }
 
 func (e *Encoder) EncodeFloat32(n float32) error {
+	if e.useCompact {
+		m := float64(n)
+		if !math.IsInf(m, 0) && !math.IsNaN(m) && float32(int64(n)) == n {
```

Comment:
Do we need `math.IsInf` and `math.IsNaN` checks here and in `EncodeFloat64`? It looks like `float32(int64(n)) == n` can't be true for Inf or NaN...

Reply:
@kuwerty gentle ping